### DISCUSSION TITLE
[SOS]数据量到百万级别时搜索不出来数据

### DISCUSSION THREAD

**Discussion by hf185** [milvus-io/milvus#12998] [Q&A and General discussion]
当数据量在几十万的时候能按向量正常搜出来数据，但是当数据导入到两百万左右时数据就搜索不出来了，请问是什么问题呢？

**Comment by yhmo** [comment]
v2.0 or v1.x?

**Reply by hf185** [reply]
2.0

**Reply by hf185** [reply]
我重新导入数据后又可以了，是否跟我第一次导入时一边插入数据一边建索引有关系呢？

**Reply by yhmo** [reply]
可能是2.0的bug，但不确定是否有修复，你用的是rc8么？

**Reply by hf185** [reply]
> 可能是2.0的bug，但不确定是否有修复，你用的是rc8么？

v2.0.0-rc8-20211104-d1f4106

**Reply by yhmo** [reply]
有一种可能是：
standalone模式下，query node和index node实际上运行在同一主机里，当数据量达到某个值时，index node开始建索引，由于建索引会大量占用CPU算力，严重影响查询性能。

**Reply by hf185** [reply]
> 

我用的是分布式版本。一开始我是插入数据的同时建立索引，这样导致数据插入两百多万的时候搜索不出来一条数据，但是后面再新增的数据就能查出来，可也只能查出来后插入的那一条。
后面我没找出来问题所在就把集合删除了再重新导入数据。并且排除了建立索引这一步，这样反而成功了，后面再新增数据也是正常的。

**Reply by yanliang567** [reply]
1. 请问Milvus在整个过程中有没有过停机操作？如果有的话，停机前milvus运行了多久，停机了多久？
2. 到200万时，搜索不出来的具体表现是卡住无返回，返回为空，还是返回的数据与期望不符？
3. build index 完成后（不论是边插入边索引，还是先插入再索引）都可以用utility.index_building_progress(collection_name)来确认一下index是否完成
4. index完成后，再load collection，如果之前有load，需要先release再load以确认load的是index

我们测试过程一般也是边建索引边插入的，暂时没有复现您的这个问题

**Reply by hf185** [reply]
> 1. 请问Milvus在整个过程中有没有过停机操作？如果有的话，停机前milvus运行了多久，停机了多久？
> 2. 到200万时，搜索不出来的具体表现是卡住无返回，返回为空，还是返回的数据与期望不符？
> 3. build index 完成后（不论是边插入边索引，还是先插入再索引）都可以用utility.index_building_progress(collection_name)来确认一下index是否完成
> 4. index完成后，再load collection，如果之前有load，需要先release再load以确认load的是index
> 
> 我们测试过程一般也是边建索引边插入的，暂时没有复现您的这个问题

1.没有停机过，用milvus-insight连接一直正常，而且还正常显示entity数量，但是同样搜索不出来数据
2. 没有卡主，响应empty result
4. 我现在是load了一次后就没有再release了，我尝试过每次查询前去load，但偶尔load要一会才能完成。我们搜索的挺频繁的，如果每次查询前load，是否更耗性能？我还没找到oad和release的最佳时机

**Reply by yanliang567** [reply]
search返回empty result应该是个issue，麻烦能复现一下，然后收集一下rootcoord，querycoord，querynode，proxy的log，到社区记一个issue吧

**Reply by hf185** [reply]
> search返回empty result应该是个issue，麻烦能复现一下，然后收集一下rootcoord，querycoord，querynode，proxy的log，到社区记一个issue吧

我之前删除了Collections后重新导入数据就正常了，我尝试下是否能复现这个问题
